### ISSUE TITLE
Use of Validation Dataset to tackle overfitting

### ISSUE THREAD

**Issue by arashidbd** [Intel-bigdata/imllib-spark#9]
Hi,
In the original FFM authors' multi-threaded implementation (https://github.com/guestwalk/libffm), there's a crucial reliance on "validation dataset" to guard against overfitting of the non-convex optimization. 

Is this implementation similar in that regard?

Thanks!
-Al

**Comment by VinceShieh** [comment]
Hi AI
sorry, I don't get your point. CV is one of the default procedures to check over-fitting of a model but that's more on the user side. On the model side, it's regularization. Not sure if that's what you are after. 
and yes, we referred to the original paper when implementing the model.

**Comment by arashidbd** [comment]
Hi Vincent,
Thanks for the reply.

Please take a look at the example here (how the training loss keeps going down after the 7th iter, even with strong regularization): https://github.com/guestwalk/libffm

Basically, FFM needs _early stopping_ with a validation set during model building. Regularization many times fails to stop overfitting with this otherwise powerful algorithm.

Thanks,
Al

**Comment by VinceShieh** [comment]
so, your question is whether we apply early-drop in our implementation? If that's your question, the answer is no. But it's a nice feature to have.

**Comment by arashidbd** [comment]
Thank you!
